paddle/fluid/pybind/imperative.cc

```diff
@@ -640,55 +640,109 @@ void BindImperative(py::module *m_ptr) {
              return TensorToPyArray(tensor, true);
            },
            R"DOC(
-        **Notes**:
-            **This API is ONLY available in Dygraph mode**
-
-        Returns a numpy array shows the value of current :ref:`api_guide_Variable_en`
 
+        Returns a numpy array shows the value of current Tensor.
+        
         Returns:
-            ndarray: The numpy value of current Variable.
+            ndarray: The numpy value of current Tensor.
 
         Returns type:
-            ndarray: dtype is same as current Variable
+            ndarray: dtype is same as current Tensor
 
         Examples:
             .. code-block:: python
 
-                import paddle.fluid as fluid
-                from paddle.fluid.dygraph.base import to_variable
-                from paddle.fluid.dygraph import Linear
+                import paddle
                 import numpy as np
-
                 data = np.random.uniform(-1, 1, [30, 10, 32]).astype('float32')
-                with fluid.dygraph.guard():
-                    linear = Linear(32, 64)
-                    data = to_variable(data)
-                    x = linear(data)
-                    print(x.numpy())
+                linear = paddle.nn.Linear(32, 64)
+                data = paddle.to_tensor(data)
+                x = linear(data)
+                print(x.numpy())
 
        )DOC")
       .def("detach",
-           [](const imperative::VarBase &self) {
-             const auto &tensor = self.Var().Get<framework::LoDTensor>();
-             PADDLE_ENFORCE_EQ(tensor.IsInitialized(), true,
-                               platform::errors::InvalidArgument(
-                                   "%s has not been initialized", self.Name()));
-             return self.NewVarBase(tensor.place(), false);
+           [](const imperative::VarBase
+                  &self) -> std::shared_ptr<imperative::VarBase> {
+             PADDLE_ENFORCE_EQ(
+                 self.Var().IsInitialized() &&
+                     (self.Var().IsType<framework::LoDTensor>() ||
+                      self.Var().IsType<framework::SelectedRows>()),
+                 true, platform::errors::InvalidArgument(
+                           "Tensor %s has not been initialized!", self.Name()));
```

Comment:
The message "Tensor %s has not been initialized!" doesn't match the requirement `self.Var().IsType<framework::LoDTensor>()  || self.Var().IsType<framework::SelectedRows>()`. Please split this `PADDLE_ENFORCE_EQ()` into two parts.

Reply:
Done